File: ClassicLib/messaging/formatting/formatter.py
"""Rust-accelerated text formatting utilities (required)."""

from __future__ import annotations

import classic_message


def strip_emoji(text: str) -> str:
    """Strip emojis from text for log safety.

    Uses required Rust implementation.

    Args:
        text: Input text possibly containing emojis.

    Returns:
        Text with emojis removed and whitespace trimmed.

    Example:
        >>> strip_emoji("Hello World!")
        'Hello  World !'

    """
    return classic_message.strip_emoji(text)


def format_log_message(content: str, details: str | None = None) -> str:
    r"""Format message content with optional details for logging.

    Strips emojis from both content and details to avoid encoding issues
    on Windows console. Uses required Rust implementation.

    Args:
        content: Main message content.
        details: Optional additional details.

    Returns:
        Formatted log message suitable for logging.

    Example:
        >>> format_log_message("Success!", "All tests passed")
        'Success!\nDetails: All tests passed'

    """
    return classic_message.format_log_message(content, details)

File: business-logic/classic-message-core/src/formatter.rs
//! Message formatting utilities preserving valid UTF-8.

/// Formats a message for logging while preserving valid UTF-8.
///
/// This function prepares a message for logging by:
/// 1. Returning content verbatim when no details are present
/// 2. Appending details verbatim as `"\nDetails: {details}"` when present
///
/// # Arguments
///
/// * `content` - The main message content.
/// * `details` - Optional additional details.
///
/// # Returns
///
/// A formatted string suitable for logging.
///
/// # Examples
///
/// ```rust
/// use classic_message_core::format_log_message;
///
/// let formatted = format_log_message("Success! ✅", Some("All tests passed 🎉"));
/// assert_eq!(formatted, "Success! ✅\nDetails: All tests passed 🎉");
/// ```
#[must_use]
pub fn format_log_message(content: &str, details: Option<&str>) -> String {
    match details {
        Some(d) => format!("{content}\nDetails: {d}"),
        None => content.to_string(),
    }
}

#[cfg(test)]
#[path = "formatter_tests.rs"]
mod tests;
